Incorrect type for spreadConstraints value in Redis chart

**Which chart**:
Redis 15.4.1 and current version

**Describe the bug**
The `spreadContraints` value is specified with a Map `{}` type, but it should be a List of Maps according to Kubernetes documentation. This is confirmed by the example in the Redis chart documentation:

```
  ## @param replica.spreadConstraints Spread Constraints for Redis&trade; replicas pod assignment
  ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
  ## E.g.
  ## spreadConstraints:
  ##   - maxSkew: 1
  ##     topologyKey: node
  ##     whenUnsatisfiable: DoNotSchedule
```

When using this as a value, the behavior is still mostly correct, but the expected warnings are generated during Golang coalescing:

> coalesce.go:200: warning: cannot overwrite table with non table for spreadConstraints (map[])

**To Reproduce**
Specify the `spreadContraints` example value in the documentation as an input.

**Expected behavior**
No warnings.

**Version of Helm**:
3.6.3

--- a/bitnami/redis/Chart.yaml
+++ b/bitnami/redis/Chart.yaml
@@ -25,4 +25,4 @@ name: redis
 sources:
   - https://github.com/bitnami/bitnami-docker-redis
   - http://redis.io/
-version: 15.6.6
+version: 15.6.7

--- a/bitnami/redis/README.md
+++ b/bitnami/redis/README.md
@@ -164,7 +164,7 @@ The command removes all the Kubernetes components associated with the chart and 
 | `master.affinity`                           | Affinity for Redis&trade; master pods assignment                                                  | `{}`                     |
 | `master.nodeSelector`                       | Node labels for Redis&trade; master pods assignment                                               | `{}`                     |
 | `master.tolerations`                        | Tolerations for Redis&trade; master pods assignment                                               | `[]`                     |
-| `master.spreadConstraints`                  | Spread Constraints for Redis&trade; master pod assignment                                         | `{}`                     |
+| `master.spreadConstraints`                  | Spread Constraints for Redis&trade; master pod assignment                                         | `[]`                     |
 | `master.lifecycleHooks`                     | for the Redis&trade; master container(s) to automate configuration before or after startup        | `{}`                     |
 | `master.extraVolumes`                       | Optionally specify extra list of additional volumes for the Redis&trade; master pod(s)            | `[]`                     |
 | `master.extraVolumeMounts`                  | Optionally specify extra list of additional volumeMounts for the Redis&trade; master container(s) | `[]`                     |
@@ -242,7 +242,7 @@ The command removes all the Kubernetes components associated with the chart and 
 | `replica.affinity`                           | Affinity for Redis&trade; replicas pods assignment                                                  | `{}`                     |
 | `replica.nodeSelector`                       | Node labels for Redis&trade; replicas pods assignment                                               | `{}`                     |
 | `replica.tolerations`                        | Tolerations for Redis&trade; replicas pods assignment                                               | `[]`                     |
-| `replica.spreadConstraints`                  | Spread Constraints for Redis&trade; replicas pod assignment                                         | `{}`                     |
+| `replica.spreadConstraints`                  | Spread Constraints for Redis&trade; replicas pod assignment                                         | `[]`                     |
 | `replica.lifecycleHooks`                     | for the Redis&trade; replica container(s) to automate configuration before or after startup         | `{}`                     |
 | `replica.extraVolumes`                       | Optionally specify extra list of additional volumes for the Redis&trade; replicas pod(s)            | `[]`                     |
 | `replica.extraVolumeMounts`                  | Optionally specify extra list of additional volumeMounts for the Redis&trade; replicas container(s) | `[]`                     |

--- a/bitnami/redis/values.yaml
+++ b/bitnami/redis/values.yaml
@@ -315,7 +315,7 @@ master:
   ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
   ##
   tolerations: []
-  ## @param master.spreadConstraints Spread Constraints for Redis&trade; master pod assignment
+  ## @param master.spreadConstraints [array] Spread Constraints for Redis&trade; master pod assignment
   ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
   ## E.g.
   ## spreadConstraints:
@@ -323,7 +323,7 @@ master:
   ##     topologyKey: node
   ##     whenUnsatisfiable: DoNotSchedule
   ##
-  spreadConstraints: {}
+  spreadConstraints: []
   ## @param master.lifecycleHooks for the Redis&trade; master container(s) to automate configuration before or after startup
   ##
   lifecycleHooks: {}
@@ -629,7 +629,7 @@ replica:
   ## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
   ##
   tolerations: []
-  ## @param replica.spreadConstraints Spread Constraints for Redis&trade; replicas pod assignment
+  ## @param replica.spreadConstraints [array] Spread Constraints for Redis&trade; replicas pod assignment
   ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/
   ## E.g.
   ## spreadConstraints:
@@ -637,7 +637,7 @@ replica:
   ##     topologyKey: node
   ##     whenUnsatisfiable: DoNotSchedule
   ##
-  spreadConstraints: {}
+  spreadConstraints: []
   ## @param replica.lifecycleHooks for the Redis&trade; replica container(s) to automate configuration before or after startup
   ##
   lifecycleHooks: {}

